test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -0,0 +1,142 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+#include "envoy/extensions/transport_sockets/tls/v3/cert.pb.h"
+
+#include "extensions/transport_sockets/tls/context_config_impl.h"
+#include "extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "test/integration/http_integration.h"
+#include "test/integration/ssl_utility.h"
+
+namespace Envoy {
+namespace {
+
+class ProxyFilterIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
```

Comment:
It was a little confusing to me that this is an HTTP integration test that actually uses the TCP proxy. Can you add a comment up top that this is a TCP proxy test in which the TLS from the client is being passed all the way through to the fake upstream, etc?

Reply:
Done